Documentation/concurrency.md

```diff
@@ -0,0 +1,27 @@
+# Concurrency
+
+In general, gRPC-go provides a concurrency-friendly API. What follows are some
+guidelines.
+
+A [ClientConn][client-conn] can safely be accessed concurrently. Using
```

Comment:
High level suggestion for organization of this (albeit small --but it may grow) doc:

\#\# Client

A \[ClientConn\]...

\#\# Streams

When using streams, one must take care to avoid calling either...

\#\# Servers

Each RPC handler attached to a registered server...

Reply:
👍